### DISCUSSION TITLE
Database new server development

### DISCUSSION THREAD

**Discussion by jo-wp** [directus/directus#10741] [Archive]
Hello,

I change my server of development windows to linux.
And an other big change is mysql to mariadb.

I have a problem when I trye: npx directus database migrate:latest 

```
alter table `directus_shares` add constraint `directus_shares_collection_foreign` foreign key (`collection`) references `directus_collections` (`collection`) on delete CASCADE - ER_CANT_CREATE_TABLE: Can't create table `demo`.`directus_shares` (errno: 150 "Foreign key constraint is incorrectly formed")
```

Do you have info for that ?

Thank you !

### To Reproduce

npx directus database migrate:latest 

### What version of Directus are you using?

9.4.0

### What version of Node.js are you using?

16.13.0

### What database are you using?

mariadb 10.3.31

### What browser are you using?

chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

locally

**Comment by rijkvanzanten** [comment]
Heya! Do you have the DB_CHARSET environment variable configured? Could you confirm the character set / collation of your existing tables? I'm assuming that a mismatch in collations used is what's causing MySQL to prevent you from creating the relation
